contracts/Pool.sol

```diff
@@ -383,30 +390,59 @@ contract Pool is PoolFDT {
         emit Cooldown(msg.sender, uint256(0));
     }
 
+    /**
+        @dev   Update the effective deposit date based on how much new capital has been added.
+               If more capital is added, the depositDate moves closer to the current timestamp.
+        @dev   It emits a `DepositDateUpdated` event.
+        @param amt Total deposit amount.
+        @param who Address of user depositing.
+    */
+    function _updateDepositDate(uint256 amt, address who) internal {
+        uint256 prevDate = depositDate[who];
+        uint256 balance = balanceOf(who);
+
+        // prevDate + (now - prevDate) * (amt / (balance + amt))
+        // NOTE: prevDate = 0 implies balance = 0, and equation reduces to now
+        uint256 newDate = (balance + amt) > 0
+            ? prevDate.add(block.timestamp.sub(prevDate).mul(amt).div(balance + amt))
+            : prevDate;
+
+        depositDate[who] = newDate;
+        emit DepositDateUpdated(who, newDate);
+    }
+
```

Comment:
No change, just moved over from `PoolLib`.

Reply:
How much change did it give you from the byte code size perspective?

Reply:
Verified diff by copying into local develop branch in PoolLib, LGTM